docs/tutorial/writing-tests/test-plan.rst

```diff
@@ -0,0 +1,383 @@
+deterministically. One of the dangers of letting Checkbox
+automatically pull resource jobs for you is that, in some situations, like
+``template-resource``, it won't do it. If you were to remove the test that
+actually pulled the resource automatically (the one that uses it as in the
+``resource`` field), you would inadvertently lose test coverage.
+
+Let's update the test plan including it:
+
+.. code-block::
+
+  unit: test plan
+  id: tutorial-extended
+  _name: Extended Tutorial Test Plan
+  bootstrap-include:
+    network_iface_info
+  include:
+    network_available_interface
+    network_available
+    network_speed certification-status=blocker
+  certification_status_overrides:
+    apply blocker to network_available
+
+When we run expand on the test plan, two important changes occur in the output:
+
+First, the resource job is no longer visible - this is expected! The bootstrap
+section of a test plan is meant to gather essential data before the main test
+execution but is not composed of actual tests, so the jobs there are excluded
+from the expand command.
+
+Second, our newly added template wasn't expanded. This happens because a
+template is expanded on the result of a resource, and only running the resource
+can give that output (that is often specific to one machine!). If we want to
+see what would the test plan expand to on the current machine, we can use
+``list-bootstrapped``:
+
+Try to run the following
+
+.. code-block::
+
+  # Note: your output will be slightly different, depending on how many ifaces you have!
+  (checkbox_venv) $ checkbox-cli list-bootstrapped com.canonical.certification::tutorial-extended
+  com.canonical.certification::network_iface_info
+  com.canonical.certification::network_available_enp2s0f0
+  com.canonical.certification::network_available_enp5s0
+  com.canonical.certification::network_available_wlan0
+  com.canonical.certification::network_available_enp7s0f3u1u2
+  com.canonical.certification::network_available
+  com.canonical.certification::network_speed
+
+
+Nested Parts
+============
+
+It is often useful to re-use the same test plan to test a functionality. This
+is for many reasons but mainly the fact that test plans are always evolving,
+adding better tests, increasing the coverage, removing old ones, and to keep
+them in sync is a very error prone chore. Checkbox has a feature to help with
+this: ``nested_part``.
+
+When a test plan has a ``nested_part``, all "parts" (jobs + other nested parts)
+are added to the test plan. Let's try this with an example. When a new test plan
+is being developed for certification purposes, one nested part is compulsory to
+include (or the submissions will be rejected): ``submission-cert-automated``.
+Let's include it in our test plan:
+
+.. code-block::
+
+  unit: test plan
+  id: tutorial-extended
+  _name: Extended Tutorial Test Plan
+  bootstrap_include:
+    network_iface_info
+  include:
+    network_available_.*
```

Comment:
You were using a template id in the previous test plan definition, and you switched to regex. Is this intentional?